lib/announcement_widget.dart

```diff
@@ -0,0 +1,267 @@
+              Utils.secondaryColor,
+              Utils.primaryColor,
+            ],
+          ),
+        ),
+        child: Stack(
+          children: [
+            Align(
+              alignment: Alignment.topCenter,
+              child: AnnouncementImage(announcement: announcement),
+            ),
+            Align(
+              alignment: Alignment.bottomCenter,
+              child: InfoSheet(announcement: announcement),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+class AnnouncementImage extends StatelessWidget {
+  const AnnouncementImage({
+    Key key,
+    @required this.announcement,
+  }) : super(key: key);
+
+  final Announcement announcement;
+
+  @override
+  Widget build(BuildContext context) {
+    return CachedNetworkImage(
+      height: MediaQuery.of(context).size.height * 0.3,
+      imageUrl: WebAPI.baseURL + announcement.poster.url,
+      placeholder: (_, __) => SizedBox(
+        height: 40,
+        width: 40,
+        child: CircularProgressIndicator(),
+      ),
+    );
+  }
+}
+
+class InfoSheet extends StatelessWidget {
+  const InfoSheet({
+    Key key,
+    @required this.announcement,
+  }) : super(key: key);
+
+  final Announcement announcement;
+
+  TextStyle _style(BuildContext context) => TextStyle(
+        color: Colors.black87,
+        fontWeight: FontWeight.w500,
+        fontSize: MediaQuery.of(context).size.aspectRatio * 35,
+      );
+
+  Future<void> _goTo(String url) async {
+    if (await canLaunch(url)) {
+      await launch(url);
+    } else {
+      throw 'Could not launch $url';
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return SizedBox(
+      height: MediaQuery.of(context).size.height,
+      child: DraggableScrollableSheet(
+        minChildSize: 0.4,
+        initialChildSize: 0.4,
+        maxChildSize: 0.6,
+        builder: (_, cont) {
+          return Container(
+            padding: EdgeInsets.only(top: Utils.padding.top),
+            decoration: BoxDecoration(
+              color: Colors.white,
+              borderRadius: BorderRadius.vertical(
+                top: Radius.circular(40),
+              ),
+            ),
+            child: ListView(
+              controller: cont,
+              physics: ClampingScrollPhysics(),
+              padding: const EdgeInsets.only(top: 12, left: 12),
+              children: [
+                Align(alignment: Alignment.center, child: DraggableIndicator()),
+                SizedBox(height: 20),
+                SelectableText(
+                  'On Behalf of: ${announcement.onBehalfOf}',
+                  style: _style(context),
+                ),
+                Divider(
+                  color: Colors.transparent,
+                  height: 10,
+                ),
+                SelectableText(
+                  'Date: ${DateFormat.yMMMMd('en_US').format(announcement.deadline)}',
+                  style: _style(context),
+                ),
```

Comment:
The deadline field in the Announcements model is not required, so if any announcement without a deadline that woud raise an error as you expect always getting a deadline 
```suggestion
                (announcement.deadline != null)
                    ? SelectableText(
                        'Date: ${DateFormat.yMMMMd('en_US').format(announcement.deadline)}',
                        style: _style(context),
                      )
                    : Container(),

```